drivers/eeprom/include/eeprom_types.h

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (C) 2018 Inria
+ *               2019 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup    drivers_eeprom
+ *
+ * @{
+ *
+ * @file
+ * @brief       Types definitions used in the EEPROM API
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ * @author      Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+ */
+
+#ifndef EEPROM_TYPES_H
+#define EEPROM_TYPES_H
+
+#include "board.h"
+#include "eeprom.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if !defined(HAVE_EEPROM_T) || defined(DOXYGEN)
+/**
+ * @brief   Device handle type for EEPROM devices
+ */
+typedef uint_fast8_t eeprom_t;
+#endif
+
+#if !defined(HAVE_EEPROM_OFF_T) || defined(DOXYGEN)
+/**
+ * @brief   Offset / size type in EEPROM devices
+ */
+typedef uint32_t eeprom_off_t;
```

Comment:
What about calling the type `eeprom_offset_t` and be explicit ? I find `eeprom_off_t` a bit misleading.

Reply:
Same for `HAVE_EEPROM_OFFSET_T` macro.

Reply:
Done